x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/utils.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import seedrandom from 'seedrandom';
+
+const DEFAULT_PARTIAL_SHUFFLE_SEED = 'seed';
+
+/**
+ * Partially shuffles an array using the Fisher-Yates algorithm.
+ *
+ * The array is shuffled in place, so that:
+ *   - the first `start` elements are kept in place;
+ *   - the elements in the slice from start to end represent the random sample;.
+ *   - the order of elements after end can be arbitrary (but the same over invocations).
+ *
+ * The result is reproducible for the given random seed.
+ *
+ * Examples:
+ *   - shuffle the whole array: partialShuffleArray(arr)
+ *   - shuffle the first 5 elements: partialShuffleArray(arr, 0, 5)
```

Comment:
What are the concrete use cases where we need to shuffle only part of the array?

Reply:
The use case is in this PR, as we do a partial shuffle for samples from index 1 to 9, since we only send 10 samples max at this moment and we want to keep the first sample.

We could theoretically shuffle all the elements from 1 until the end of the array and throw out the extra elements, but that would be additional unnecessary work.

Reply:
So it's for efficiency purposes, I see. Thanks!